pkg/scheduler/objects/queue.go

```diff
@@ -246,9 +259,36 @@ func (sq *Queue) applyConf(conf configs.QueueConfig) error {
 	}
 
 	sq.properties = conf.Properties
+	sq.limits = conf.Limits
 	return nil
 }
 
+func (sq *Queue) GetConfigLimits() []configs.Limit {
+	var limit []configs.Limit
+	// get the limit for the parent first
+	if sq.parent != nil {
+		limit = sq.parent.GetConfigLimits()
+	}
+	return sq.internalGetConfigLimits(limit)
+}
+
+func (sq *Queue) internalGetConfigLimits(parentLimit []configs.Limit) []configs.Limit {
+	sq.RLock()
+	defer sq.RUnlock()
+	// no limit set in parent queue, not even for root
+	if parentLimit == nil {
+		if sq.limits == nil {
+			return nil
+		}
+		return sq.limits
+	}
+	// parent limit set, no queue limit return parent
+	if sq.limits == nil {
+		return parentLimit
+	}
+	return sq.limits
```

Comment:
Couldn't this be simplified?

Eg. 
```if sq.limits != nil {
  return. sq.limits
}

return parentLimit
```

So we return the current queues's limit if not nil, otherwise we just pass the parentLimit no matter what.